Libraries/Opc.Ua.Security.Certificates/X509Crl/RevokedCertificate.cs

```diff
@@ -0,0 +1,130 @@
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ * OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * The complete license agreement can be found here:
+ * http://opcfoundation.org/License/MIT/1.00/
+ * ======================================================================*/
+
+using System;
+using System.Security.Cryptography.X509Certificates;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Opc.Ua.Security.Certificates
+{
+    /// <summary>
+    /// Represents a revoked certificate in the
+    /// revoked certificates sequence of a CRL.
+    /// </summary>
+    /// <remarks>
+    /// CRL fields -- https://tools.ietf.org/html/rfc5280#section-5.1
+    /// 
+    ///    ...
+    ///    revokedCertificates     SEQUENCE OF SEQUENCE  {
+    ///        userCertificate         CertificateSerialNumber,
+    ///        revocationDate          Time,
+    ///        crlEntryExtensions      Extensions OPTIONAL
+    ///                              -- if present, version MUST be v2
+    ///                            }  OPTIONAL,
+    ///   ...
+    ///</remarks>
+    public class RevokedCertificate
+    {
+        /// <summary>
+        /// Construct revoked certificate with serialnumber,
+        /// actual UTC time and the CRL reason.
+        /// </summary>
+        /// <param name="serialNumber">The serial number</param>
+        /// <param name="crlReason">The reason for revocation</param>
+        public RevokedCertificate(string serialNumber, CRLReason crlReason)
+            : this(serialNumber)
+        {
+            CrlEntryExtensions.Add(X509Extensions.BuildX509CRLReason(crlReason));
+        }
+
+        /// <summary>
+        /// Construct revoked certificate with serialnumber,
+        /// actual UTC time and the CRL reason.
+        /// </summary>
+        /// <param name="serialNumber">The serial number</param>
+        /// <param name="crlReason">The reason for revocation</param>
+        public RevokedCertificate(byte[] serialNumber, CRLReason crlReason)
+            : this(serialNumber)
+        {
+            if (crlReason != CRLReason.Unspecified)
+            {
+                CrlEntryExtensions.Add(X509Extensions.BuildX509CRLReason(crlReason));
+            }
+        }
+
+        /// <summary>
+        /// Construct minimal revoked certificate
+        /// with serialnumber and actual UTC time.
+        /// </summary>
+        /// <param name="serialNumber"></param>
+        public RevokedCertificate(string serialNumber) : this()
+        {
+            UserCertificate = serialNumber.FromHexString().Reverse().ToArray();
```

Comment:
This looks like the asymmetric signature I mentioned for `FromHeyString` extension method